backend/src/main/java/kernel360/techpick/feature/folder/model/FolderProvider.java

```diff
@@ -39,11 +41,30 @@ public Folder findRecycleBin(Long userId) {
 		return folderRepository.findRecycleBinByUserId(userId);
 	}
 
+	public Folder findRoot(Long userId) {
+		return folderRepository.findRootByUserId(userId);
+	}
+
 	public void deleteById(Long folderId) {
 		folderRepository.deleteById(folderId);
 	}
 
 	public boolean existsByUserIdAndName(Long userId, String name) {
 		return folderRepository.existsByUserIdAndName(userId, name);
 	}
+
+	// 현재 폴더가 어떤 BasicFolder에 속해있는지 반환하는 함수
+	// TODO: Spring Data Jpa에서 재귀쿼리를 제공하지 않아 반복조회로 해결.. QueryDSL 도입 후 리팩토링 필요
+	public FolderType findParentBasicFolder(Long folderId) {
+
+		EnumSet<FolderType> basicFolderTypes = FolderType.getBasicFolderTypes();
+		Folder currentFolder = this.findById(folderId);
+
+		while (basicFolderTypes.contains(currentFolder.getFolderType())) {
+			currentFolder = this.findById(currentFolder.getId());
```

Comment:
p2: parentFolder의 id로 해야 하는것같은데 확인 부탁
```suggestion
			currentFolder = this.findById(currentFolder.getParentFolder().getId());
```

Reply:
수정완료